clang/test/CIR/CodeGen/array-unknown-bound.cpp

```diff
@@ -4,7 +4,7 @@ extern int table[];
 // CHECK: cir.global external @table = #cir.const_array<[#cir.int<1> : !s32i, #cir.int<2> : !s32i, #cir.int<3> : !s32i]> : !cir.array<!s32i x 3>
 
 int *table_ptr = table;
-// CHECK: cir.global external @table_ptr = #cir.global_view<@table> : !cir.ptr<!s32i>
+// CHECK: cir.global external @table_ptr = #cir.global_view<@table> : !cir.ptr<!cir.array<!s32i x 3>>
```

Comment:
this test now looks better - we replace the type for `table` and `table_ptr` has proper type now

Reply:
Hmm, I see your point but from the language consistency perspective this is probably not right? If I write `int *table_ptr` I'm expecting this to be a `!cir.ptr<!s32i>`, not a `!cir.ptr<!cir.array<!s32i x 3>>`. For the purposes of needed computation, why looking at the original `@table` type isn't enough, global view basically operates like a an attribute level type cast here.

Reply:
fixed!